lib/buildkite/test_collector/test_links_plugin/reporter.rb

```diff
@@ -0,0 +1,60 @@
+# frozen_string_literal: true
+
+module Buildkite::TestCollector::TestLinksPlugin
+  class Reporter
+    RSpec::Core::Formatters.register self, :dump_failures
+
+    def initialize(output)
+      @output = output
+    end
+
+    def dump_failures(notification)
+      # Do not display summary if no failed examples
+      return unless notification.failed_examples.present?
+
+      # Check if a Test Analytics token is set
+      return unless Buildkite::TestCollector.api_token
+
+      metadata = fetch_metadata
+
+      # return if metadata was not fetched successfully
+      return if metadata.nil?
+
+      # return if suite url is nil
+      return if metadata['suite_url'].nil?
+
+      @output << "\n\nTest Analytics failures:\n\n"
+
+      @output << notification.failed_examples.map do |example|
+        failed_example_output(example, metadata['suite_url'])
+      end.join("\n")
+
+      @output << "\n\n"
+    end
+
+    private
+
+    def generate_scope_name_digest(scope, name)
+      Digest::SHA256.hexdigest(scope.to_s + name.to_s)
```

Comment:
I'm a bit surprised that `scope` and `name` here aren't strings already? They seem like they should be from lines 42-32?

Reply:
Do we know for sure these will be strings? I see that we are converting to strings here https://github.com/buildkite/buildkite/blob/main/app/models/analytics/execution/batch_creator.rb#L165

So maybe there is a chance that it could contain numbers?

Reply:
Interesting...I reckon that it's done that way in the execution batch creator to handle nil cases? Because test collector ruby always sends values for scope and name, but other test collectors / users won't necessarily

Reply:
I had a bit of a dig through the rspec docs and it does seem to indicate they should be strings: https://rubydoc.info/gems/rspec-core/RSpec/Core/Example 

Anyways it's not a huge deal - happy for you to merge as is, I was just a little surprised to see those `to_s` calls there!

Reply:
Ohh interesting! That makes sense about them coming from other collectors. I will do a few tests with some random non-string test names and if its all good I will remove!

Reply:
Tested with numbers and it converts to string, will remove the `to_s`